astropy/units/tests/test_equivalencies.py

```diff
@@ -998,3 +999,17 @@ def test_pprint():
         "<tr><td>Ci</td><td>3.7e+10 / s</td><td>curie</td></tr>"
         "<tr><td>Hz</td><td>1 / s</td><td>Hertz, hertz</td></tr></table>"
     )
+
+
+def test_spectral_density_factor_deprecation():
+    with pytest.warns(
+        AstropyDeprecationWarning,
+        match=(
+            r'^"factor" was deprecated in version 7\.0 and will be removed in a future '
+            r"version\. \n        Use `wav` as a `Quantity` instead\.$"
```

Comment:
```suggestion
            r"version\. \n        Use wav as a Quantity instead\.$"
```

Reply:
I had a closer look at the warning message and noticed that in the automatically generated part the name `factor` is highlighted using double-quotes (`"`). It is best to use consistent quoting throughout the message.